### ISSUE TITLE
Week 5 Linear Regression

### ISSUE THREAD

**Issue by RamiChumber** [weisongwen/AAE4011-S22526#7]
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
# Load the dataset
data = pd.read_csv('./test.csv')
# Check the column names in the dataset
print(data.columns)
# Assume the dataset has two columns X and y, prepare the data
X = data['x'].values.reshape(-1, 1) # Feature variable
y = data['y'].values # Target variable
# Create and fit the linear regression model
lin_reg = LinearRegression()
lin_reg.fit(X, y)

# Make predictions
y_pred = lin_reg.predict(X)

# Calculate model score
r_squared = lin_reg.score(X, y)

# Create visualization
plt.figure(figsize=(10, 6))

# Plot the actual data points
plt.scatter(X, y, color='blue', s=100, label='Actual data', alpha=0.7, edgecolors='black')

# Plot the regression line
plt.plot(X, y_pred, color='red', linewidth=2, label='Linear regression line')

# Customize the plot
plt.xlabel('X (Feature)', fontsize=12, fontweight='bold')
plt.ylabel('y (Target)', fontsize=12, fontweight='bold')
plt.title('Linear Regression Visualization', fontsize=14, fontweight='bold')
plt.legend(fontsize=11, loc='upper left')
plt.grid(True, alpha=0.3)

# Add statistics text box to the plot
stats_text = f'Slope: {lin_reg.coef_[0]:.4f}\nIntercept: {lin_reg.intercept_:.4f}\nR² Score: {r_squared:.4f}'
plt.text(0.98, 0.05, stats_text, transform=plt.gca().transAxes, 
         fontsize=11, verticalalignment='bottom', horizontalalignment='right',
         bbox=dict(boxstyle='round', facecolor='wheat', alpha=0.8))

# Show the plot
plt.tight_layout()
plt.savefig('linear_regression_plot.png', dpi=300, bbox_inches='tight')
print("Plot saved as 'linear_regression_plot.png'")
plt.show()

<img width="2964" height="1764" alt="Image" src="https://github.com/user-attachments/assets/77be4c29-e585-4b99-88c9-34546807161f" />
